src/shared/Validation.js

```diff
@@ -25,9 +25,80 @@ export const validateOnInput = (
   const formatErrors = sectionFormatErrors.slice();
 
   switch (fieldType) {
-    case 'number': {
+    case 'int16': {
       const idx = formatErrors.indexOf(FORM_ERRORS.INVALID_FORMAT);
-      if (input && !validator.isInt(input)) {
+      const validInt = validator.isInt(input);
+
+      let validIntType = true;
+      if (parseInt(input, 10) < -32768 || parseInt(input, 10) > 32767) {
+        validIntType = false;
+      }
+
+      if ((input && !validInt) || (input && !validIntType)) {
```

Comment:
If `input` is falsy, then `validator.isInt` will either throw an Error or return false. Either way, this check seems redundant.

Reply:
Fixed